integrationtests/self/handshake_test.go

```diff
@@ -51,10 +51,18 @@ var _ = Describe("Handshake tests", func() {
 	}
 
 	Context("Version Negotiation", func() {
+		var supportedVersions []protocol.VersionNumber
+
+		BeforeEach(func() {
+			supportedVersions = protocol.SupportedVersions
+			protocol.SupportedVersions = append(protocol.SupportedVersions, []protocol.VersionNumber{7, 8, 9, 10}...)
+		})
+
+		AfterEach(func() {
+			protocol.SupportedVersions = supportedVersions
+		})
+
```

Comment:
Can you run this with the race detector to make sure there aren't any leftovers after the tests that might still access those?

Reply:
I ran the self-tests more than 10 times with race detector in randomized order, and everything seems ok.